internal/alloy/logging/logger.go

```diff
@@ -124,11 +143,23 @@ func (l *Logger) Update(o Options) error {
 	}
 	l.writer.Set(newWriter)
 
+	// Build all our deferred handlers
+	if l.deferredSlog != nil {
+		l.deferredSlog.buildHandlers(nil)
+	}
 	// Print out the buffered logs since we determined the log format already
 	for _, bufferedLogChunk := range l.buffer {
-		// the buffered logs are currently only sent to the standard output
-		// because the components with the receivers are not running yet
-		slogadapter.GoKit(l.handler).Log(bufferedLogChunk...)
+		if len(bufferedLogChunk.kvps) > 0 {
+			// the buffered logs are currently only sent to the standard output
+			// because the components with the receivers are not running yet
+			slogadapter.GoKit(l.handler).Log(bufferedLogChunk.kvps...)
+		} else {
+			// We now can check to see if if our buffered log is at the right level.
+			if l.level.Level() <= bufferedLogChunk.record.Level {
```

Comment:
nit: I'd rather this use `bufferedLogChunk.handler.Enabled(context.Background(), bufferedLogChunk.record.Level)`; to use the Handler API as it was intended to be used.